### ISSUE TITLE
referrerPolicy is unset if json is used

### ISSUE THREAD

**Issue by whs** [sindresorhus/ky#439]
Testing in browsers, it seems that referrerPolicy will be unset if request is copied in certain ways:

```
// Test that request initialization with referrerPolicy works
> new Request('https://example.com', {referrerPolicy: "unsafe-url"}).referrerPolicy
"unsafe-url" 

// Test that copying the request do indeed copy the referrerPolicy
> new Request(new Request('https://example.com', {referrerPolicy: "unsafe-url", method:"POST", body: "test"})).referrerPolicy
"unsafe-url" 

// However, if specifying body, the referrerPolicy would be unset
> new Request(new Request('https://example.com', {referrerPolicy: "unsafe-url", method:"POST"}), {body: "test"}).referrerPolicy
"" 
```

The above console snippet is verified on

* Mozilla/5.0 (X11; Linux x86_64; rv:101.0) Gecko/20100101 Firefox/101.0
* Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36

According to [fetch spec](https://fetch.spec.whatwg.org/#dom-request) when the second param is specified a few parameters are reset (bullet point 13) - including referrerPolicy.

In https://github.com/sindresorhus/ky/blob/dc76278c8a6dc73843e1a6e86acba639e2bf63e4/source/core/Ky.ts#L181 when `json` option is provided, the request is copied with body added. Since `referrerPolicy` is not copied by the browser's API it will be lost.

**Comment by sholladay** [comment]
Well, that is... annoying. Seems like we should be able to copy over the referrer policy manually, though.

```js
this.request = new globalThis.Request(this.request, {body: this._options.body, referrerPolicy: this.request.referrerPolicy}); 
```

**Comment by whs** [comment]
The spec (linked in OP) also mention origin, referrer, referrerPolicy, mode (only if it was 'navigate', I doubt that it is valid for our use case?) as all the fields cleared during copying

**Comment by sholladay** [comment]
Closing as I believe this is fixed by PR #600.
